addons/sale_timesheet/models/account_move.py

```diff
@@ -37,47 +39,71 @@ def _compute_timesheet_count(self):
         for invoice in self:
             invoice.timesheet_count = mapped_data.get(invoice.id, 0)
 
+    def _compute_has_timesheet_portal(self):
+        Timesheet = self.env['account.analytic.line']
+        initial_domain = Timesheet._timesheet_get_portal_domain()
+        for invoice in self:
+            invoices = self.env['account.move'].sudo().search(['|', ('name', 'ilike', invoice.name), ('id', 'ilike', invoice.name)])
```

Comment:
Don't call ``search`` in a for loop (twice), make a global call outside of the loop and then map the data accordingly to prevent unecessary requests to the DB.

``('id', 'ilike', invoice.name)`` what do you expect over here ? ``id`` is an integer, ``invoice.name`` is a char.